src/exchanges/cache.ts

```diff
@@ -145,19 +146,23 @@ const afterQuery = (
   resultCache: ResultCache,
   operationCache: OperationCache
 ) => (response: OperationResult) => {
-  const {
-    operation: { key },
-    data,
-  } = response;
+  const { operation, data } = response;
+
   if (data === undefined) {
     return;
   }
 
-  resultCache.set(key, response);
+  resultCache.set(operation.key, {
+    ...response,
+    operation: {
+      ...operation,
+      context: { ...operation.context, cacheHit: true },
```

Comment:
So for the normalised cache can we introduce more of an enum, which will cover all cases:

```ts
type CacheOutcome = 'MISS' | 'PARTIAL' | 'HIT';
```